backend/src/test/scala/com/softwaremill/adopttapir/test/Describers.scala

```diff
@@ -0,0 +1,25 @@
+package com.softwaremill.adopttapir.test
+
+import com.softwaremill.adopttapir.starter.StarterDetails
+
+import java.io.{PrintWriter, StringWriter}
+import scala.util.Using
+
+object Describers {
```

Comment:
JFYI - not required any changes. 

This approach is similar to TypeClasses. 

```
trait Describer[A] {
  def describe(value: A): String
}

object DescriberInstances {
  implicit val detailsDescriber = new Describer[StarterDetails] {
    override def describe(details: StarterDetails): String =
      s"${details.serverEffect}/${details.serverImplementation}/docs=${details.addDocumentation}/metrics=${details.addMetrics}" +
        s"/json=${details.jsonImplementation}/scalaVersion=${details.scalaVersion}/builder=${details.builder}"
  }

  implicit val throwableDescriber = new Describer[Throwable] {
    override def describe(e: Throwable): String = {
      Using.Manager { use =>
        val sw = use(new StringWriter())
        val pw = use(new PrintWriter(sw))
        e.printStackTrace(pw)
        sw.toString
      }.get
    }
  }
}

object DescriberSyntax {
  implicit class DescriberOps[A](value: A) {
    def describe(implicit d: Describer[A]): String = d.describe(value)
  }
}
```

And as it is similar to cats Show https://typelevel.org/cats/typeclasses/show.html 

You can just use build-in class type and only provide expected instances. The rest - syntax will be for free. 

```
object ShowInstances {
  implicit val dShow = Show.show[StarterDetails](details =>
    s"${details.serverEffect}/${details.serverImplementation}/docs=${details.addDocumentation}/metrics=${details.addMetrics}" +
      s"/json=${details.jsonImplementation}/scalaVersion=${details.scalaVersion}/builder=${details.builder}"
  )

  implicit val throwableShow = Show.show[Throwable] { e =>
    Using.Manager { use =>
      val sw = use(new StringWriter())
      val pw = use(new PrintWriter(sw))
      e.printStackTrace(pw)
      sw.toString
    }.get

  }
}

```
and then use show type class whenever you want

Reply:
Long story short: You don't have to create another abstraction - cats have build-in solution for U.